### PR TITLE
Fix products search results order in SearchProductsHandler

### LINKED ISSUE #17282: Finish @todo in 1.7.7 migrated code
We put some `@todo` in 1.7.7 migrated code to be handled later.

Later is now 💪 !

For each `@todo`, we must
- [remove it if irrelevant
or
- fix it
or
- answer it
or
- [create an issue about it

### TODO List

**In BO Theme JS:**
- [x] js/pages/order/create/customer-manager.js
- [x] js/pages/order/message/order-view-page-messages-handler.js

**In Adapter namespace:**
* [x] src/Core/Domain/Address/Command/SetRequiredFieldsForAddressCommand - #17526
* [x]  src/Adapter/Order/CommandHandler/AddCartRuleToOrderHandler.php - #17348
* [ ]  src/Adapter/Order/CommandHandler/AddProductToOrderHandler.php
* [x]  src/Adapter/Order/CommandHandler/ChangeOrderCurrencyHandler.php - #17672
* [ ]  src/Adapter/Order/CommandHandler/DeleteProductFromOrderHandler.php
* [ ]  src/Adapter/Order/CommandHandler/UpdateProductInOrderHandler.php
* [ ]  src/Adapter/Order/QueryHandler/GetOrderForViewingHandler.php
* [x]  src/Adapter/Order/Refund/OrderRefundCalculator.php - #17850
* [x]  src/Adapter/Order/Refund/OrderSlipCreator.php - `associated with mailer. As we don't have a new mailer yet, these todo's should be left for future`
* [x]  src/Adapter/Order/Refund/VoucherGenerator.php - `associated with mailer. As we don't have a new mailer yet, these todo's should be left for future`
* [x]  src/Adapter/Product/QueryHandler/SearchProductsHandler.php - #17736 

**In Core namespace:** 
* [ ]  src/Core/Grid/Data/Factory/CatalogPriceRuleGridDataFactory.php
* [ ]  src/Core/Grid/Definition/Factory/Monitoring/EmptyCategoryGridDefinitionFactory.php

**In PrestaShopBundle:**

* [ ]  src/Core/Grid/Data/Factory/CatalogPriceRuleGridDataFactory.php
* [ ]  src/Core/Grid/Definition/Factory/Monitoring/EmptyCategoryGridDefinitionFactory.php
* [ ]  src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php
* [ ]  src/PrestaShopBundle/Resources/views/Admin/Sell/Order/Order/Blocks/Create/cart.html.twig

**All checked items only have its designated pull requests or answers, it might not be merged yet.**

### PR DESCRIPTION
**Pull request by zuk3975**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.7.x
| Description?  | Fix todo `sort products alphabetically` in SearchProductsHandler. Removed id index in product result array which caused results list order mess. Updated usage in order-view.js. Order create page seems to be working as expected. Both searches should return results in alphabetical order.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | part of #17282
| How to test?  | Check that order view page - product searching and adding works as expected. Order create page - product searching and adding works as expected

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/17736)

### CODE REVIEW

**Review by Progi1984 — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
PrettyCI complains => probably false positive, see https://github.com/PrestaShop/PrestaShop/issues/17744
Travis behat tests fail => this one must be checked

**Review by matthieu-rolland — DISMISSED** [review]

File: tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php [hunk 1]
```diff
@@ -98,22 +99,24 @@ public function createEmptyCartForCustomer(string $cartReference, string $custom
...
+        $products = $this->getQueryBus()->handle(new SearchProducts($productName, 1, Context::getContext()->currency->iso_code));
 
-        if (!$productId) {
-            throw new RuntimeException(sprintf('Product with name "%s" does not exist', $productName));
+        if (empty($products)) {
+            throw new RuntimeException(sprintf('Product with name "%s" was not found', $productName));
         }
 
+        /** @var FoundProduct $product */
+        $product = $products[0];
```
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php]:
```suggestion
        $product = reset($products);
```

File: tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php [hunk 2]
```diff
@@ -538,14 +539,16 @@ public function checkProductStockDifference(int $productDifference, string $fact
...
 
-        if (!$productId) {
-            throw new RuntimeException('Product with name "%s" does not exist', $productName);
+        if (empty($products)) {
+            throw new RuntimeException(sprintf('Product with name "%s" was not found', $productName));
         }
 
-        return (int) $productId;
+        /** @var FoundProduct $product */
+        $product = $products[0];
```
Comment (Progi1984) [inline tests/Integration/Behaviour/Features/Context/Domain/OrderFeatureContext.php]:
```suggestion
        $product = reset($products);
```

File: tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php (hunk 1 above)
Comment (zuk3975) [reply tests/Integration/Behaviour/Features/Context/Domain/CartFeatureContext.php]: didn't know about this method :open_mouth:  looks better indeed, thanks :smile_cat:

**Comment by zuk3975** [comment]
@Progi1984 thanks, addressed.

**Review by matks — DISMISSED** [review]

**Comment by matks** [comment]
Good to me, waiting for another review

**Review by PierreRambaud — DISMISSED** [review]

**Comment by boubkerbribri** [comment]
@zuk3975 please resolve conflicts

**Comment by zuk3975** [comment]
> @zuk3975 please resolve conflicts

done

**Review by matks — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @zuk3975

**Pull request merged**